### ISSUE TITLE
Task overriding broken in 0.7.11

### ISSUE THREAD

**Issue by dfreeman** [machty/ember-concurrency#81]
In 0.7.10, overriding a task definition in an `extend` call worked as I'd expect a normal property override to, but with the getter changes in 0.7.11 it now throws `TypeError: Cannot set property go of [object Object] which has only a getter`.

```js
const MyThing = Ember.Object.extend({
  go: task(function*() { return 'yes'; })
});

const MyOtherThing = MyThing.extend({
  go: task(function*() { return 'actually, no'; })
});
```

**Comment by machty** [comment]
I'm heading out of town for a week and a half and probably won't be able to address this in a reasonable amount of time; any interest in contributing a fix?

I'm guessing it'd just require tweaking the configuration we provide to `defineProperty` here: https://github.com/machty/ember-concurrency/blob/master/addon/-task-property.js#L420-L424

**Comment by machty** [comment]
Also does it need to be writable, or just configurable?

**Comment by sukima** [comment]
Um, not sure, but based on the descriptions in MDN both seemed needed to make the property function as expected a _normal_ property would.

> **configurable**: true if and only if the type of this property descriptor may be changed and if the property may be deleted from the corresponding object.
>
> **writable**: true if and only if the value associated with the property may be changed with an assignment operator.
>
> _~ [MDN: defineProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty)_

**Comment by camhux** [comment]
Just adding a no-op `set` trap seems to fix this:
```
let cp = this;
Object.defineProperty(proto, taskName, {
  get() {
    return cp.get(this, taskName);
  },
  set() {/*trap*/},
});
```
This looks like it shouldn't work ([this line](https://github.com/emberjs/ember.js/blob/aeeddf7c00b42a340494465252481b5414d87009/packages/ember-metal/lib/properties.js#L159), in fact, has no effect with this setter) but it nonetheless causes a test case to pass. As far as I can tell, thanks to Ember delegating to the descriptor's `setup` method [here](https://github.com/emberjs/ember.js/blob/aeeddf7c00b42a340494465252481b5414d87009/packages/ember-metal/lib/properties.js#L161), the descriptor ends up properly set/reset when overridden.

I have this change and test case on a fork. I don't mind opening a PR for review with the caveat that I don't fully grok the Ember property mechanisms involved, which is not a feeling I like when submitting code.

**Comment by sukima** [comment]
That feels quite fishy. Does PR #82 also fix this? The PR feels more standard compliant then having a no-op setter which, if it were me, I would scratch my head and go on faith that a comment `// this works somehow` to be true.

**Comment by machty** [comment]
I believe @camhux is correct. ember-metal does a property set when merging mixins, so we have write an empty setter so that things don't below up. I've implemented this in #83
